File: app/src/main/java/com/beacat/calendar/ladycal/DeleteAllPeriodsTask.java
package com.beacat.calendar.ladycal;

import android.content.Context;
import android.os.AsyncTask;

import com.tyczj.extendedcalendarview.PeriodDatabase;

/**
 *
 */
public class DeleteAllPeriodsTask extends AsyncTask<Void, Void, Void> { //params, progress, result

    private Context context; // needed to insert the new entry into the db

    public DeleteAllPeriodsTask(Context context) {
        super();
        this.context = context;
    }

    @Override
    protected Void doInBackground(Void... params) {
        PeriodDatabase db = PeriodDatabase.getInstance(context);
        db.resetDB();
        return null;
    }

    @Override
    protected void onPostExecute(Void v){
    }
}

File: app/src/main/java/com/beacat/calendar/ladycal/DeleteAllPeriodsTask.kt
package com.beacat.calendar.ladycal

import android.content.Context
import android.os.AsyncTask

import com.tyczj.extendedcalendarview.PeriodDatabase

/**
 *
 */
class DeleteAllPeriodsTask(
    private val context: Context // needed to insert the new entry into the db
) : AsyncTask<Void, Void, Void>() { //params, progress, result

    override fun doInBackground(vararg params: Void): Void? {
        val db = PeriodDatabase.getInstance(context)
        db.resetDB()
        return null
    }

    override fun onPostExecute(v: Void?) {
    }
}
